### PR TITLE
Add subcontrol details

### PR DESCRIPTION
**Pull request by minhaj10p**

### CODE REVIEW

File: generator/intersection.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 )
```
Comment (mohuk) [inline generator/intersection.go]: File is named `intersection.go`. Its more of a mapper than an intersection.

File: generator/intersection.go [hunk 2]
```diff
@@ -37,24 +37,82 @@ func CreateCatalogsFromProfile(profile *profile.Profile) []*catalog.Catalog {
...
 
-func mapSubControls(mappedCatalogs *catalog.Catalog, calls []profile.Call) {
-	for i, g := range mappedCatalogs.Groups {
-		for j, control := range g.Controls {
-			for _, call := range calls {
-				if strings.ToLower(control.Id) == strings.ToLower(getControlIDFromSubControl(call.SubcontrolId)) {
-					mappedCatalogs.Groups[i].Controls[j].Subcontrols = append(mappedCatalogs.Groups[i].Controls[j].Subcontrols, catalog.Subcontrol{
-						Id: call.SubcontrolId,
-					})
+func getGroupID(g *catalog.Group) (string, error) {
```
Comment (mohuk) [inline generator/intersection.go]: As discussed, this logic is NIST-800-53 specific, it should be generalized. Let's discuss this in person.

File: generator/mapper.go [hunk 3]
```diff
@@ -92,21 +58,36 @@ func getMappedCatalogControlsFromImport(importedCatalog *catalog.Catalog, profil
 		}
 		//Append controls to the new group if matches
 		for _, catalogControl := range group.Controls {
-			for _, z := range profileImport.Include.IdSelectors {
+			for controlIndex, z := range profileImport.Include.IdSelectors {
```
Comment (asadullah-yousuf-10p) [inline generator/mapper.go]: What is `z`? @minhaj10p

File: generator/mapper.go [hunk 4]
```diff
@@ -92,21 +58,36 @@ func getMappedCatalogControlsFromImport(importedCatalog *catalog.Catalog, profil
...
-					newGroup.Controls = append(newGroup.Controls, catalog.Control{
+					newControl := catalog.Control{
 						Id:          catalogControl.Id,
 						Class:       catalogControl.Class,
 						Title:       catalogControl.Title,
 						Subcontrols: []catalog.Subcontrol{},
-					})
+					}
+					//For subcontrols, find again in entire profile
+					for _, sc := range catalogControl.Subcontrols {
```
Comment (asadullah-yousuf-10p) [inline generator/mapper.go]: `sc` should be replaced with more meaningful name.

Comment (mohuk) [reply generator/mapper.go]: I believe thats how they code in golang, brevity over clarity

Comment (minhaj10p) [reply generator/mapper.go]: yeah, you'll see a lot of "m" letter in orca as well. 
still changing it though

File: generator/generator_test.go [hunk 5]
```diff
@@ -121,7 +121,10 @@ func TestCreateCatalogsFromProfile(t *testing.T) {
 			},
 		},
 	}
-	x := CreateCatalogsFromProfile(&p)
+	x, err := CreateCatalogsFromProfile(&p)
```
Comment (asadullah-yousuf-10p) [inline generator/generator_test.go]: Rename this to something meaningful `x`.

**Pull request merged**